support/signal.c

```diff
@@ -175,30 +260,36 @@ default_handler (int signum)
 		signal_info* h = &signals [i];
 		if (mph_int_get (&h->signum) != signum)
 			continue;
+
 		mph_int_inc (&h->count);
+
+		if (!acquire_pipelock_handler (&h->pipelock))
+			continue; // Teardown is occurring on this object, no one to send to.
+
 		fd = mph_int_get (&h->write_fd);
-		if (fd > 0) {
+		if (fd > 0) { // If any pipe exists to write to
 			int j,pipecounter;
-			char c = signum;
-			pipecounter = mph_int_get (&h->pipecnt);
+			char c = signum;                         // (Actual value written to pipe is discarded)
```

Comment:
Needless comment; what does this comment add?

We could instead do `char c = 0; /* value meaningless */`, but again, that doesn't necessarily add much.

What it *does* add is noise to the patch.